### PR TITLE
server: Add uptime status var and statistics

### LINKED ISSUE #8842: support global status like 'uptime'
## Feature Request

**Is your feature request related to a problem? Please describe:**

TiDB does not support the status variable 'uptime'.  MySQL dbas use it idiomatically:
* I restart the server, I check it is the correctly rebooted.
* My client disconnected.  Was it because of a server crash?  check uptime.
* has the server freshly started?  = slow performance (not applicable to tidb, since it's stateless).

So it is not critical but helps improve 'comfort'.

**Describe the feature you'd like:**

`SHOW GLOBAL STATUS like 'Uptime'` should report uptime in seconds.

**Describe alternatives you've considered:**

N/A

**Teachability, Documentation, Adoption, Migration Strategy:**

Compatibility with MySQL.

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #8842

Problem Summary:

- The uptime in the `mysql.ComStatistics` was always 0
- There was no `Uptime` status var, causing compatibility issues with the status command in https://github.com/dbcli/mycli

<pre>mysql&gt; status
--------------
...
Uptime:			7 min 2 sec

Threads: 0  Questions: 0  Slow queries: 0  Opens: 0  Flush tables: 0  Open tables: 0  Queries per second avg: 0.000
--------------

mysql&gt; SHOW GLOBAL STATUS LIKE &apos;Uptime&apos;;
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| Uptime        | 423   |
+---------------+-------+
1 row in set (0.03 sec)
</pre>

And MySQL Shell:
```
sql> \status
MySQL Shell version 8.0.27

...
Uptime:                       21 min 34.0000 sec

Threads: 0  Questions: 0  Slow queries: 0  Opens: 0  Flush tables: 0  Open tables: 0  Queries per second avg: 0.000
```

(patched) MyCli:
```
MySQL root@127.1:(none)> status
--------------
mycli 1.24.1, running on CPython 3.10.0
...
| Uptime:              | 22 min 41 sec                                                                                                          |
+----------------------+------------------------------------------------------------------------------------------------------------------------+
--------------
Time: 0.031s
```

Related: https://github.com/dbcli/mycli/pull/1014

This matches the information in the HTTP status API (`/info`, `start_timestamp`) and the `information_schema.CLUSTER_INFO`. Note that *all* of these might report the current time instead of the real startup time if no PD/etcd is available.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [x] Changes MySQL compatibility

### Release note

```release-note
A global status variable for `Uptime` was added.
```

### CODE REVIEW

**Review by morgo — APPROVED** [review]
LGTM. I think some automated scripts might have some trouble with `SHOW GLOBAL STATUS like 'Uptime'` because they probably don't consider case. But we'll have to fix that separately.

**Review by mjonss — APPROVED** [review]
LGTM

**Comment by bb7133** [comment]
Please add a test case for your change, thanks.

**Comment by dveeden** [comment]
> Please add a test case for your change, thanks.

Done.

Note that the previous `info.StartTimestamp` had a value of `1282967700000` when in the mock/failpoint. This has three digits more than a regular UNIX timestamp. To me it looks like this should be a regular UNIX timestamp in seconds.

When running a `tiup playground`:
```
$ etcdctl get --print-value-only /topology/tidb/127.0.0.1:4000/info | jq '.start_timestamp'
1636957939

$ curl -s http://127.0.0.1:10080/info | jq '.start_timestamp'
1636957939
```

File: domain/infosync/info_test.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func TestTopology(t *testing.T) {
 
 	topology, err := info.getTopologyFromEtcd(ctx)
 	require.NoError(t, err)
-	require.Equal(t, int64(1282967700000), topology.StartTimestamp)
```
Comment (bb7133) [inline domain/infosync/info_test.go:73]: Why change the mock value? Is the previous too large to be meanningful?

Comment (dveeden) [reply domain/infosync/info_test.go:73]: I'll change this to to use the original timestamp, but with `000` removed.

See also https://github.com/pingcap/tidb/pull/29790#issuecomment-979486931

**Comment by dveeden** [comment]
@bb7133 PTAL

File: server/stat_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tidb/domain/infosync"
+	"github.com/pingcap/tidb/session"
+	"github.com/pingcap/tidb/store/mockstore"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUptime(t *testing.T) {
+	t.Parallel()
```
Comment (bb7133) [inline server/stat_test.go:30]: Please remove `Parallel()`, as planned in https://github.com/pingcap/tidb/issues/30822.

Comment (dveeden) [reply server/stat_test.go]: Done. PTAL.

**Review by bb7133 — APPROVED** [review]

**Comment by yahonda** [comment]
Thanks for introducing this feature. It would contribute supporting `innotop` which uses `show status like 'Uptime'`.

https://github.com/innotop/innotop/blob/945cd0a972cefa0e2ebc5cfd622db935f31def1a/innotop#L7653

**Pull request merged**